sdk/src/Services/DynamoDBv2/Custom/DocumentModel/TableBuilder.cs

```diff
@@ -0,0 +1,260 @@
+        private HashSet<string> _attributesAlreadyProcessed; 
+
+        /// <summary>
+        /// Creates a builder object to construct a <see cref="Table"/>
+        /// </summary>
+        /// <param name="ddbClient">Client to use to access DynamoDB.</param>
+        /// <param name="tableName">Table name</param>
+        public TableBuilder(IAmazonDynamoDB ddbClient, string tableName) :
+             this(ddbClient, tableName, DynamoDBEntryConversion.CurrentConversion, false, null)
+        {
+        }
+
+        /// <summary>
+        /// Creates a builder object to construct a <see cref="Table"/>
+        /// </summary>
+        /// <param name="ddbClient">Client to use to access DynamoDB.</param>
+        /// <param name="tableName">Table name</param>
+        /// <param name="conversion">Conversion to use for converting .NET values to DynamoDB values.</param>
+        /// <param name="isEmptyStringValueEnabled">If the property is false, empty string values will be interpreted as null values.</param>
+        /// <param name="metadataCachingMode">The document API relies on an internal cache of the DynamoDB table's metadata to construct and validate 
+        /// requests. This controls how the cache key is derived, which influences when the SDK will call 
+        /// <see cref="IAmazonDynamoDB.DescribeTable(string)"/> internally to populate the cache.</param>
+        public TableBuilder(IAmazonDynamoDB ddbClient, string tableName, DynamoDBEntryConversion conversion, bool isEmptyStringValueEnabled, MetadataCachingMode? metadataCachingMode)
+            : this (ddbClient, new TableConfig(tableName, conversion, Table.DynamoDBConsumer.DocumentModel, null, isEmptyStringValueEnabled, metadataCachingMode))
+        {
+        }
+
+        /// <summary>
+        /// Creates a builder object to construct a <see cref="Table"/>
+        /// </summary>
+        /// <param name="ddbClient">Client to use to access DynamoDB.</param>
+        /// <param name="config">Configuration to use for the table.</param>
+        public TableBuilder(IAmazonDynamoDB ddbClient, TableConfig config)
+        {
+            _table = new Table(ddbClient, config);
+            _table.ClearTableData(); // initializes internal collections
+            _attributesAlreadyProcessed = new HashSet<string>();
+        }
+
+        /// <inheritdoc/>
+        public Table Build()
+        {
+            if (_table.HashKeys.Count == 0)
+            {
+                throw new ArgumentOutOfRangeException("A partition key definition is required, call AddHashKey before Build.");
+            }
+
+            return _table;
+        }
+
+        /// <inheritdoc/>
+        public ITableBuilder AddHashKey(string hashKeyAttribute, DynamoDBEntryType type)
+        {
+            if (string.IsNullOrEmpty(hashKeyAttribute))
+            {
+                throw new ArgumentNullException(nameof(hashKeyAttribute), "The name of the partition key attribute is required.");
```

Comment:
Be consistent and use hash key.

Reply:
Fixed in https://github.com/aws/aws-sdk-net/compare/74020f5167a3bf2e54dd53bdeba482f965e43dd2..abcc18e865203cd0aa124c2c175f0dc798d5be93